runtime/src/bank/fee_distribution.rs

```diff
@@ -50,24 +69,56 @@ impl Bank {
                             },
                         ));
                     }
-                }
-                Err(_) => {
-                    error!(
-                        "Burning {} fee instead of crediting {}",
-                        deposit, self.collector_id
-                    );
-                    datapoint_error!(
-                        "bank-burned_fee",
-                        ("slot", self.slot(), i64),
-                        ("num_lamports", deposit, i64)
-                    );
-                    burn += deposit;
+                    Err(err) => {
+                        error!(
+                            "Burning {} fee instead of crediting {} due to {err}",
+                            deposit, self.collector_id
+                        );
+                        datapoint_error!(
+                            "bank-burned_fee",
+                            ("slot", self.slot(), i64),
+                            ("num_lamports", deposit, i64)
+                        );
+                        burn += deposit;
+                    }
                 }
             }
             self.capitalization.fetch_sub(burn, Relaxed);
         }
     }
 
+    // Deposits fees into a specified account and if successful, returns the new balance of that account
+    fn deposit_fees(
+        &self,
+        pubkey: &Pubkey,
+        fees: u64,
+        options: DepositFeeOptions,
+    ) -> Result<u64, DepositFeeError> {
+        let mut account = self.get_account_with_fixed_root(pubkey).unwrap_or_default();
```

Comment:
why not fail directly here rather than do the whole footgun-prone `unwrap_or_default()` thing?

Reply:
Because it's fine if the account doesn't exist as long as the fees top up the balance to at least rent exempt.